### DISCUSSION TITLE
Mana Shield

### DISCUSSION THREAD

**Discussion by teddybeard78** [diasurgical/DevilutionX#5933] [Q&A] [answered]
Hi. Is there any way to reverse the Mana Shield fix? I have always considered the bug, that made your character immune to stun and knockback with the shield on, to be a good feature. It makes sense to me that it is the mage's alternative to armor, and that it should drain their energy before anything could harm them at all. Maybe others see it differently, but is there, perhaps, a way to make some small change in the files, and make it an alternative offshoot for those who do like it that way? I'm not enough of a programmer to do it myself, but maybe it's not a problem for someone who knows the how? It would be much appreciated.

**Comment by DakkJaniels** [comment]
Since it was in the code before, of course it's possible. I doubt this would be reimplemented in DevilutionX, however.

**Reply by teddybeard78** [reply]
Maybe make it a file, with which you could overwrite something, or maybe it's something as simple as changing a few strings in some file(s), that you can use notepad for.

**Reply by DakkJaniels** [reply]
That's not the issue, it's adding code and toggles for turning on bugs, which is unlikely to happen in DevilutionX. Mana shield disabling combat mechanics is silly, it's basically just playing with god mode on.

**Reply by teddybeard78** [reply]
That is your opinion, but since you aren't interested in the topic, why reply at all? It's not as if I'm making you play the game this way, shrug.

**Reply by StephenCWills** [reply]
It may not be the answer you want to hear, but he's just trying to tell you that you're unlikely to convince anyone to do this for you. It was not an intentional mechanic, regardless of how you saw it.

**Reply by qndel** [reply]
https://github.com/diasurgical/devilution here's a mod that restores all original bugs

**Reply by teddybeard78** [reply]
The way I see it, is that the people who are renovating the game, don't have to follow a rigid policy of ironing out every single bug, and making the game play some specific, 'intended' way. Like I'd said, I'm one of those people who see abilities such as the Mana Shield as a question of character style, rather than a problem of balance or game difficulty. Yes, the Sorcerer is a stronger character, but I see no need to balance different characters against each other, and the over-all difficulty of the game can be tuned separately from specific abilities.

Another useful 'bug' of the game is the ability to refresh shop inventories through reloading saves. This ended up being a way to circumvent the restrictive approach that the developers took in regard to equipment availability. Even with this 'bug', you could play through the entire game multiple times before ever seeing some of the equipment or spells that are possible.

**Reply by DakkJaniels** [reply]
That's fine. There is nothing stopping you from learning how to modify the game and making it however you want. You can either try and find the commit that fixed the bug and change it back, or you can find out where in the code the character goes into hit recovery/knockback and just make it not happen if the character has mana shield.

**Comment by AJenbo** [comment, marked as answer]
Things like what you are requesting might seem simple on the surface, but can quickly grow complicated, how would it work for multiplayer, who is going to maintain it when new versions come out? Will we remember to test both settings? The added complexity is why we try to somewhat limit the scope of things. Thank you for your understanding.

**Reply by teddybeard78** [reply]
Sure. This is the sort of reply I was expecting. I only asked in case this didn't require you going out of your way, or if it was so simple that I could do it on my end.

**Reply by AJenbo** [reply]
I took a quick look and it will require someone moderately familiar with C++. The code for manashield was drastically changed over multiple iterations to resolve the many issues with it. So there isn't a simple change to reference which still applies. You can of course run older versions of DevilutionX from before the change.

**Reply by DakkJaniels** [reply]
I would just comment out anything related to the knockback effect and hit recovery (the end of StartPlrHit and the Monster melee and missile attack functions, I think). No reason why any of this needs to be balanced.

**Reply by teddybeard78** [reply]
I'll try that. Or try an earlier version if I fail.
